src/mlpack/methods/ann/augmented/tasks/add_impl.hpp

```diff
@@ -0,0 +1,128 @@
+/**
+ * @file add_impl.hpp
+ * @author Konstantin Sidorov
+ *
+ * Implementation of AddTask class
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_AUGMENTED_TASKS_ADD_IMPL_HPP
+#define MLPACK_METHODS_AUGMENTED_TASKS_ADD_IMPL_HPP
+
+#include "add.hpp"
+
+namespace mlpack {
+namespace ann /* Artificial Neural Network */ {
+namespace augmented /* Augmented neural network */ {
+namespace tasks /* Task utilities for augmented */ {
+
+AddTask::AddTask(const size_t bitLen) : bitLen(bitLen)
+{
+  if (bitLen <= 0)
+  {
+    std::ostringstream oss;
+    oss << "AddTask::AddTask(): binary length (" << bitLen << ") "
+        << "is not positive!"
+        << std::endl;
+    throw std::invalid_argument(oss.str());
+  }
+}
+
+const void AddTask::Generate(arma::field<arma::mat>& input,
+                             arma::field<arma::mat>& labels,
+                             const size_t batchSize,
+                             bool fixedLength)
+{
+  input = arma::field<arma::mat>(batchSize);
+  labels = arma::field<arma::mat>(batchSize);
+  size_t sizeA = bitLen, sizeB = bitLen;
+  for (size_t i = 0; i < batchSize; ++i)
+  {
+    if (!fixedLength)
+    {
+      // Generate random uniform length from [2..maxLength].
+      sizeA = mlpack::math::RandInt(2, bitLen + 1);
+      sizeB = mlpack::math::RandInt(2, bitLen + 1);
```

Comment:
Well, since we emit least significant bits first, we can simply add two `1`'s after sampling a random binary vector: before the separator (msb of first number) and in the very end of the sequence (msb of second number).